lib/model/model.go

```diff
@@ -1234,6 +1234,10 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 	m.remoteFolderStates[deviceID] = states
 	m.pmut.Unlock()
 
+	m.evLogger.Log(events.ClusterConfigReceived, map[string]string{
+		"device": deviceID.String(),
+	})
```

Comment:
I am not ok with `map[string]string` over `map[string]interface{}` for a type that might be extended with more (potential non-string) fields. And I am also inclined to require a concrete type, there's now been two instances of this messy map-string-interface-construction making problems. A concrete type avoids that entirely and I don't see how it's that complex. I can provide a patch if you'd like.

Reply:
I reverted to that because it panicked with `map[string]interface{}` and I didn't want to hurry out a solution where I don't fully understand the consequences.  As long as we don't change the Event API serialization, we can switch to whatever internal type anytime, no?

I think switching to a dedicated struct is best done in a separate change, ideally for *all* event types where the current fragile construction with typecasting to string maps is used.  Maybe Go 1.18 generics would help in that regard, and I'll happily investigate such a bigger solution, but just not as part of this change, which is mostly a fix / different approach for new functionality added in this release cycle.

Reply:
Generics won't help here that I can see. I agree with Simon that introducing a new, better, way makes sense already at this stage. The change for everything historical is going to be messy, and there's no reason to continue building brokenness until then.

Reply:
And while we can indeed change it later as long as structure exposed to APIs doesn't change, such changes are exactly the problem: If you change the type on the map you need to remember updating every place you do type assertions. With a stuct you just add the new field and all type assertions keep working.

Reply:
Better now?  The struct is currently defined in `lib/model`, similar to `FolderSummaryEventData`.  Not sure if it should better live in `lib/events`?

Reply:
Thank you!  
I think lib/events should be about the event handling, not needing to know what the actual events are. Thus having it in model seems correct to me.